rrs/views.py

```diff
@@ -0,0 +1,1005 @@
+            return True
+        elif isinstance(self.ver, int) and isinstance(other.ver, int):
+            return False
+
+    def __repr__(self):
+        return repr(self.group)
+
+
+
+class RecipeUpgradeDetail():
+    title = None
+    version = None
+    maintplan_name = None
+    release_name = None
+    milestone_name = None
+    date = None
+    maintainer_name = None
+    is_recipe_maintainer = None
+    commit = None
+    commit_url = None
+    upgrade_type = None
+    group = None
+
+    def __init__(self, title, version, maintplan_name, release_name, milestone_name, date,
+            maintainer_name, is_recipe_maintainer, commit, commit_url, upgrade_type,
+            group):
+        self.title = title
+        self.version = version
+        self.maintplan_name = maintplan_name
+        self.release_name = release_name
+        self.milestone_name = milestone_name
+        self.date = date
+        self.maintainer_name = maintainer_name
+        self.is_recipe_maintainer = is_recipe_maintainer
+        self.commit = commit
+        self.commit_url = commit_url
+        self.upgrade_type = upgrade_type
+        self.group = group
+
+def _get_recipe_upgrade_detail(maintplan, recipe_upgrade):
+    release_name = ''
+    milestone_name = ''
+    recipe_maintainer_history = None
+
+    release = Release.get_by_date(maintplan, recipe_upgrade.commit_date)
+    if release:
+        release_name = release.name
+        milestone = Milestone.get_by_release_and_date(release,
+                recipe_upgrade.commit_date)
+        if milestone:
+            milestone_name = milestone.name
+            recipe_maintainer_history = RecipeMaintainerHistory.get_by_end_date(
+                recipe_upgrade.recipesymbol.layerbranch,
+                milestone.end_date)
+
+    is_recipe_maintainer = False
+    maintainer_name = ''
+    if not recipe_upgrade.maintainer is None:
+        maintainer_name = recipe_upgrade.maintainer.name
+
+        if not recipe_maintainer_history is None and \
+            RecipeMaintainer.objects.filter(maintainer__name
+            = maintainer_name, history = recipe_maintainer_history) \
+            .count() > 0:
+            is_recipe_maintainer = True
+
+    commit_date = recipe_upgrade.commit_date.date().isoformat()
+    commit = recipe_upgrade.sha1[:10]
+    commit_url = recipe_upgrade.recipesymbol.layerbranch.commit_url(recipe_upgrade.sha1)
+
+    rud = RecipeUpgradeDetail(recipe_upgrade.title, recipe_upgrade.version, \
+            maintplan.name, release_name, milestone_name, commit_date, maintainer_name, \
+            is_recipe_maintainer, commit, commit_url, recipe_upgrade.upgrade_type, \
+            recipe_upgrade.group)
+
+    return rud
+
+class RecipeDetailView(DetailView):
+    model = RecipeSymbol
+
+    def get_queryset(self):
+        self.maintplan_name = self.kwargs['maintplan_name']
+        return super(RecipeDetailView, self).get_queryset()
+
+    def get_context_data(self, **kwargs):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 20 other functions. By comparison, 99% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 800-809 into a separate function.